docs/89-NEWS.md

```diff
@@ -1,10 +1,186 @@
+   GTK.
+ * Clients now have a `requests_no_titlebar` property to expose when a client
+   has client side titlebars (also known as decorations and CSD)
+ * The hotkey popup now has a `show_awesome_keys` option.
+ * The `awful.widget.prompt` now has more of the `awful.prompt` constructor
+   arguments.
+ * It is now possible to set a list of layouts per tag instead of a single
+   global one.
+ * There is now a `awful.layout.get_tag_layout_index()` function to get the
+   index of the current layout in the global layout list
+   (`awful.layout..layouts`)
+
+### Better DPI handling
+
+The screen now has a read/write `dpi` property and `awful.screen.set_auto_dpi_enabled(true)`
+can be used to automatically set the DPI for many Awesome elements. Please
+note that it is not backward compatible and breaks many widget. As AwesomeWM
+always used pixels as the de-facto metric for sizes, enabling `anto_dpi` will
+break most existing configs. However, for people who use such setup, it might be
+worth speding some time to fix their config.
+
+### Extendable `awful.rules` providers and better `awful.spawn` functions
+
+There is two new functions called `awful.rules.add_rule_source` and
+`awful.rules.remove_rule_source`. They allow to create a dependency graph
+for where a rule comes from and which provider has the priority when setting it.
+
+Previously, there was the normal properties,
+`awful.rules.high_priority_properties` and `awful.rules.delayed_properties`.
+This didn't scale and could not represent all corner cases. Those table still
+exist and are still honored, but there is now a system that can handle the full
+complexity of the property priority graph.
+
+Making use of this by default is `awful.spawn`. The reliability of the attaching
+properties to the `spawn` calls has been improved. On top of this, 3 new
+functions have been added
+
+ * `awful.spawn.once`
+ * `awful.spawn.single_instance`
+ * `awful.spawn.raise_or_spawn`
+
+They allow to specify a command can only have one running instance. This works
+across restart too, so all hacks to handle restarting Awesome are no longer
+required.
+
+Note that the `client.startup_id` isn't supported by all applications and the
+Linux specific workaround is recommended to improve the reliability of the
+`awful.spawn` functions.
+
+### A brand new keygrabber API
+
+The `keygrabber` module API was rebuilt from scratch to make it more usable.
+The previous API was very low level, very close to how it actually work, but
+was disconnected from how keygrabbers are used in a window manager. Getting
+anything done with the previous API required a lot of boilerplate code and had
+many corner cases to handle. The new API has built-in support for the most
+common use cases and is fully declarative.
+
+### A new GTK color palette based theme
+
+A new theme has been added. It reads the GTK theme colors and use them in the
+`wibar`, `menu` and `titlebar`. It helps create an uniform look and feel
```

Comment:
```suggestion
`wibar`, `menu` and `titlebar`. It helps creating an uniform look and feel
```

Reply:
`uniform` uses `a` rather than `an`. `create a uniform` is the correct grammar.